### PR TITLE
Load `environment` properties

### PR DESCRIPTION
**Pull request by mosteo**
Per #345, this PR implements the loading part of these properties. They can be inspected as any other property with `alr show`.

Properties can be retrieved in code with the new `Alire.Releases.Environment` function.

### CODE REVIEW

File: doc/catalog-format-spec.rst [hunk 1]
```diff
@@ -285,6 +285,32 @@ entries:
...
 
+* ``environment``: optional table used to modify environment variables that
+  will apply at build time. Variables and values are specified with the form
+  ``VARIABLE.<action> = "value"``, where ``<action>`` is one of ``append``,
+  ``prepend``, or ``set``. For instance:
+
+  .. code-block:: toml
+
+   [environment]
+   C_INCLUDE_PATH.append = "/usr/include/something"
```
Comment (Fabien-Chouteau) [inline doc/catalog-format-spec.rst:296]: Is the case maintained here? `PATH` vs `path`.

File: doc/catalog-format-spec.rst [hunk 2]
```diff
@@ -285,6 +285,32 @@ entries:
...
+   PATH.prepend = "$CRATE_ROOT/bin"
+   MSYS.set = "$MSYS_ROOT"
+
+  Other environment variables may appear in the value, and will be replaced.
+  Furthermore, a few predefined variables are provided by Alire:
+
+  * ``$CRATE_ROOT`` is the deployment directory of the crate.
```
Comment (Fabien-Chouteau) [inline doc/catalog-format-spec.rst]: We should discuss the predefined value format. I think this `${CRATE}` will be easier to handle and less error prone.

File: doc/catalog-format-spec.rst (hunk 1 above)
Comment (mosteo) [reply doc/catalog-format-spec.rst:296]: Yes, TOML keys are case sensitive and I did no modification on these.

File: doc/catalog-format-spec.rst (hunk 2 above)
Comment (mosteo) [reply doc/catalog-format-spec.rst]: Sure. At the moment there's no code about that anyway, so it's a matter of expressing what we want in the documentation. I too like better with `{}`.

**Review by Fabien-Chouteau — APPROVED** [review]

**Pull request merged**